test/DynamicDependency/Powershell/Test-MsixDynamicDependency.Tests.ps1

```diff
@@ -0,0 +1,325 @@
+# Copyright (c) Microsoft Corporation and Contributors.
+# Licensed under the MIT License.
+
+# Test Framework is Pester => https://pester.dev/docs/quick-start
+# NOTE: Powershell 5.1.22621.1778 includes module Pester version 3.4.0
+# NOTE: Powershell 7.3.6 includes module Pester version version 3.4.0
+
+Set-StrictMode -Version 3.0
+
+$ErrorActionPreference = "Stop"
+
+# Errors
+$E_INVALIDARG = 0x80070057
+
+# Repository paths
+$root = (Get-Item $PSScriptRoot).parent.parent.parent.FullName
+$dev = Join-Path $root 'dev'
+$test = Join-Path $root 'test'
+$buildOutput = Join-Path $root 'BuildOutput\Debug\x64'
+
+# Test package(s)
+$packageName = 'WindowsAppRuntime.Test.DynDep.Fwk.Widgets'
+$packageFamilyName = 'WindowsAppRuntime.Test.DynDep.Fwk.Widgets_8wekyb3d8bbwe'
+$packageFullName = 'WindowsAppRuntime.Test.DynDep.Fwk.Widgets_1.2.3.4_neutral__8wekyb3d8bbwe'
+$packageMsixDir = 'Framework.Widgets'
+$packageMsixFilename = 'Framework.Widgets.msix'
+$packageMsixPath = Join-Path $buildOutput $packageMsixDir
+$packageMsix = Join-Path $packageMsixPath $packageMsixFilename
+
+# Track resources to ensure we clean up
+$packageDependencyIds = [ordered]@{}
+$packageDependencyContexts = [ordered]@{}
+
+Function LogComment
+{
+    param([string]$message)
+
+    "       $message" | Write-Host
+}
+
+Function Trace
+{
+    param([string]$message)
+
+    Write-Host $message -ForegroundColor Magenta
+}
+
+Function TraceEnter
+{
+    param([string]$message)
+
+    Trace " Test: $message"
+}
+
+Function RemoveTestPackages
+{
+    Write-Host "Removing $packageName..." -ForegroundColor Cyan
+
+    # Remove our package(s) in case they were previously installed and incompletely removed
+    Get-AppxPackage $packageName | Remove-AppxPackage
+    $p = Get-AppxPackage $packageName
+    if ($p -ne $null)
+    {
+        Write-Error $p
+        Write-Error "Remove/Get-AppxPackage result not expected"
+        Exit 1
+    }
+}
+
+Function AddTestPackages
+{
+    Write-Host "Adding $packageMsix..." -ForegroundColor Cyan
+
+    # Install our needed package(s)
+    if (-not(Test-Path -Path $packageMsix -PathType Leaf))
+    {
+        Write-Error "$($packageMsix) not found"
+        Exit 1
+    }
+    Add-AppxPackage $packageMsix
+    $p = Get-AppxPackage $packageName
+    if (($p -eq $null) -Or ($p.PackageFullName -ne $packageFullName))
+    {
+        Write-Error $p
+        Write-Error "Get-AppxPackage result not expected"
+        Exit 1
+    }
+    if ($p.PackageFullName -ne $packageFullName)
```

Comment:
$p.PackageFullName -ne $packageFullName

Reply:
No, they're different things. This checks if the package is registered for the user. The above checks if the specified .msix file exists

When run the test emits

```
Adding D:\source\repos\windowsappsdk2\BuildOutput\Debug\x64\Framework.Widgets\Framework.Widgets.msix...
```

making it easy to tell the difference :-)